### PR TITLE
bugfix: don't remove GlobalSession when retry rollback or retry commit timeout 

### LINKED ISSUE #5295: 事务回滚重试超时后，DB会话存储的模式下会将全局事务删除掉，接下来还有修改事务状态的逻辑
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

版本：1.6.1 
`DefaultCoordinator#handleRetryRollbacking`方法中，当遇到事务回滚重试 发生了超时，会包含以下逻辑：
1. DB模式下，会从global_table中删除此事务记录（`removeGlobalSession(...)`方法中会删除）
2. 将此事务状态变更为 `RollbackRetryTimeout`
疑问DB层面已删除，此处修改状态什么作用？

在 `DefaultCoordinator#handleRetryCommitting`也存在同样的情况。

### Ⅱ. Describe what happened
如下图所示
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

针对这个疑问跟社区的 [【a364176773】](https://github.com/a364176773) 进行探讨，初步结论是 可按照Seata后续的能力规划对此处进行修改：

*  如此上下文中重试回滚超时异常类的事务，不再由每秒执行一次的定时任务来自动删除
*  把这类异常事务记录保留下来，之后在控制台侧增加功能来管控这类异常事务。

那么当下对应的修改可为：

* 将上边的`removeGlobalSession`逻辑删除掉。

### Ⅵ. Environment:

- JDK version : 8
- Seata version:  1.6.1
- OS :
- Others:

### LINKED ISSUE #5297: MySQLUndoUpdateExecutor中UPDATE_SQL_TEMPLATE的注释不正确
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

在1.6.1 版本中`MySQLUndoUpdateExecutor#UPDATE_SQL_TEMPLATE`这个字段的注释是
```
UPDATE a SET x = ?, y = ?, z = ? WHERE pk1 in (?) pk2 in (?)
```
实际sql的格式并非如此

### Ⅱ. Describe what happened

构建where条件的源码如下：
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

这个注释应该修改为：
```
 UPDATE a SET x = ?, y = ?, z = ? WHERE pk1 =? and pk2 =? 
```

### Ⅴ. Anything else we need to know?

五

### Ⅵ. Environment:

- JDK version :
- Seata version:  1.6.1
- OS :
- Others:

### PR DESCRIPTION
**Pull request by ZhangShiYeChina**
### Ⅰ. Describe what this PR did

1. Correct the comment on MySQLUndoUpdateExecutor#UPDATE_SQL_TEMPLATE (#5297)
2. #5295

### Ⅱ. Does this pull request fix one issue?

fixes #5297 
fixes #5295

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
add change in develop.md not inn 1.6.1.md

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
 public class MySQLUndoUpdateExecutor extends AbstractUndoExecutor {
 
     /**
-     * UPDATE a SET x = ?, y = ?, z = ? WHERE pk1 in (?) pk2 in (?)
+     * UPDATE a SET x = ?, y = ?, z = ? WHERE pk1 =? and pk2 =?
```
Comment (wangliang181230) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java:40]: 强迫症表示，等号后面加个空格吧。

**Review by wangliang181230 — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java (hunk 1 above)
Comment (slievrly) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java:40]: This will be fix in the next pr.

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**